crates/core/src/db/relational_db.rs

```diff
@@ -419,6 +571,20 @@ impl RelationalDB {
         }
     }
 
+    /// Decide based on the `committed_state.next_tx_offset`
+    /// whether to request that the [`SnapshotWorker`] in `self` capture a snapshot of the database.
+    ///
+    /// Actual snapshotting happens asynchronously in a Tokio worker.
+    fn maybe_do_snapshot(&self, tx_data: &TxData) {
+        if let Some(snapshot_worker) = &self.snapshot_worker {
+            if let Some(tx_offset) = tx_data.tx_offset() {
```

Comment:
I know it's been discussed, but maybe it can be explained in the commentary: why are we snapshotting independent of the durable tx offset?

Reply:
Added comment.

Reply:
Great, thanks!